### DISCUSSION TITLE
What triggers a session's history sync, and is there a way to stop that?

### DISCUSSION THREAD

**Discussion by BatmanAoD** [nushell/nushell#10541] [Q&A] [answered]
I'm used to having a separate history for each running shell session, unless I explicitly use `history -r` to load the persistent history from disk.

It seems that Nu's behavior is to occasionally synchronize a running shell's history with the on-disk history, leading to fairly confusing behavior: I expect the up-arrow (or `k` in normal Vi mode) to iterate through the most recent commands in the current shell, but occasionally it's interleaved with history from my other shell sessions.

Is there a way to only have history be synchronized when the shell is closed and/or with an explicit history command?

**Comment by fdncred** [comment, marked as answer]
set your history to use sqlite and set history isolation to true. i think that's as close as we get right now.

**Reply by BatmanAoD** [reply]
Thanks, that seems like it's exactly what I wanted! (It would still be nice to have an explicit "sync un-isolated history" command, and it looks like there are still some issues with history isolation, but this is definitely an improvement.)
